pkg/rehearse/jobs.go

```diff
@@ -628,6 +628,23 @@ func (e *Executor) submitRehearsals() ([]*pjapi.ProwJob, error) {
 	var pjs []*pjapi.ProwJob
 
 	for _, job := range e.presubmits {
+		// Remove configresolver flags from ci-operator jobs
+		if len(job.Spec.Containers) > 0 && len(job.Spec.Containers[0].Command) > 0 && job.Spec.Containers[0].Command[0] == "ci-operator" {
+			var newArg []string
+			for _, arg := range job.Spec.Containers[0].Args {
+				ignore := false
+				for _, ignoredArg := range []string{"-resolver-address", "-org", "-repo", "-branch", "-variant"} {
+					// Handle both single and double dash forms of go flags
+					if strings.HasPrefix(arg, ignoredArg) || strings.HasPrefix(arg, fmt.Sprintf("-%s", ignoredArg)) {
+						ignore = true
+					}
+				}
+				if !ignore {
+					newArg = append(newArg, arg)
```

Comment:
This does not handle the `--org value` case (where the value is passed as a following arg) which would be a pitfall. It looks like we do not use that form in our jobs but I still think we should handle that correctly to avoid surprises.